funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -747,17 +778,38 @@ def start(self, poll_period=None):
 
             self.total_pending_task_count -= dispatched_task
 
+            # Task cancel is high priority, so we'll process all requests
+            # in one go
+            while True:
+                try:
+                    manager, task_id = self.task_cancel_running_queue.get(block=False)
+                    logger.debug(f"[MAIN] Task:{task_id} on manager:{manager} is now CANCELLED while running")
+                    cancel_message = pickle.dumps(('TASK_CANCEL', task_id))
+                    self.task_outgoing.send_multipart([manager, b'', cancel_message])
+
+                except queue.Empty:
+                    break
+
             for manager in task_dispatch:
                 tasks = task_dispatch[manager]
                 if tasks:
+                    if self.task_cancel_pending_trap:
+                        for task in tasks:
+                            task_id = task["task_id"]
+                            if task_id in self.task_cancel_pending_trap:
+                                logger.warning(f"[MAIN] Task:{task_id} CANCELLED before launch")
+                                cancel_message = pickle.dumps(('TASK_CANCEL', task_id))
+                                self.task_outgoing.send_multipart([manager, b'', cancel_message])
```

Comment:
what is this trying to do differently than the cancel_running_queue code? in both cases the manager is told to cancel the task with the same message?